FFCDemoPaymentService/Messaging/MessageConfig.cs

```diff
@@ -0,0 +1,40 @@
+using Microsoft.Azure.ServiceBus.Primitives;
+
+namespace FFCDemoPaymentService.Messaging
+{
+    public class MessageConfig
+    {
+        private TokenProvider _tokenProvider;
+        public bool UseTokenProvider { get; set; }
+        public string ScheduleQueueName { get; set; }
+        public string PaymentQueueName { get; set; }
+        public string MessageQueueHost { get; set; }
+        public string MessageQueuePreFetch { get; set; }
+        public string MessageQueueUser { get; set; }
+        public string MessageQueuePassword { get; set; }
+        public string MessageQueueEndPoint
+        {
+            get
+            {
+                return $"sb://{MessageQueueHost}/";
+            }
+        }
+        public string ConnectionString
+        {
+            get
+            {
+                return $"Endpoint={MessageQueueEndPoint};SharedAccessKeyName={MessageQueueUser};SharedAccessKey={MessageQueuePassword}";
+            }
+        }
+        public TokenProvider TokenProvider {
+            get
+            {
+                if (_tokenProvider == null) {
+                    _tokenProvider = TokenProvider.CreateManagedIdentityTokenProvider();
+                }
```

Comment:
You can replace this with the null coalescing assignment operator, `??=`, rather than the if statement.

ie

`tokenProvider ??= TokenProvider.CreateManagedIdentityTokenProvider();`

If `tokenProvider` is not null it just returns the existing value, if it is it replaces it with the managed identity option.

Reply:
Done.